packages/fxa-settings/src/components/ConnectedServices/index.tsx

```diff
@@ -55,9 +61,136 @@ export const ConnectedServices = () => {
     setErrorText('Sorry, there was a problem refreshing the recovery key.');
     alertBar.show();
   });
-
   const showMobilePromo = !sortedAndUniqueClients.filter(isMobileDevice).length;
 
+  // TODO: should this view name be the exact same as in old-settings? "settings.clients.disconnect"?
+  const viewName = 'settings.clients.disconnect';
+
+  // The Confirm Disconnect modal is shown when a user clicks 'Sign Out' on a sync service.
+  // It asks the user to confirm they want to disconnect, and answer a survey question explaining
+  // why they are disconnecting.
+  const [
+    confirmDisconnectModalRevealed,
+    revealConfirmDisconnectModal,
+    hideConfirmDisconnectModal,
+  ] = useBooleanState();
+  const [disconnectingClient, setDisconnectingClient] = useState<
+    AttachedClient
+  >();
+  const [disconnectReason, setDisconnectReason] = useState<string>();
+  const [submitStartTime, setSubmitStartTime] = useState<number>();
+
+  // After the user confirms they want to disconnect from sync in Confirm Disconnect modal,
+  // if their reason was a lost/stolen device, or a suspicious device, then we show them
+  // an informative modal with some advice on next steps to take.
+  const [
+    adviceModalRevealed,
+    revealAdviceModal,
+    hideAdviceModal,
+  ] = useBooleanState();
+
+  const clearDisconnectingState = (
+    errorMessage?: string,
+    error?: ApolloError | undefined
+  ) => {
+    hideConfirmDisconnectModal();
+    // TODO: if I uncomment these lines, stuff breaks. What's the right way to coordinate
+    // hanging on to enough local state until we're done? Maybe not pass around the AttachedClient,
+    // but instead pass around copies of the data? Like the name, clientId, lastAccessTime, etc?
+    //setDisconnectingClient(undefined);
+    //setDisconnectReason(undefined);
+    //setSubmitStartTime(0);
```

Comment:
What was breaking?  I tried uncommenting those lines and didn't see any obvious failures.

Reply:
Oh, interesting! I wasn't able to trigger the secondary modal when I eagerly cleared the state. I'll give it another shot